### DISCUSSION TITLE
gitpod openvscode VS code-server

### DISCUSSION THREAD

**Discussion by its-dibo** [gitpod-io/openvscode-server#99] [General] [answered]
how does it differ from [code-server](https://github.com/cdr/code-server)

**Comment by mbtools** [comment]
This https://coder.com/pricing ?

**Reply by its-dibo** [reply]
I already use it with google cloud shell for free
the price is for hosting it like gitpod does.
but you can use [their repo](https://github.com/cdr/code-server) to install it and use it for free

be sure that we talk about `code-server` not `coder` which are different
see the last question of FAQ in the [pricing page](https://coder.com/pricing)

**Comment by SlavikCA** [comment]
Here is my summary of differences:

- code-server support auth (protect the editor with password) while OpenVSCode doesn't
- OpenVSCode installs extensions from open-vsx, while code-server is in the process of switching to open-vsx
- TAB is working in the code-server's terminal. Doesn't work in the OpenVSCode. This difference may seem small, but significant to my day-to-day use
- I figured how code-server handles user account in Docker, but haven't figure out yet how OpenVSCode does it in the Docker

**Comment by pawlean** [comment, marked as answer]
My colleague @mikenikles explained this really well, so I'll share what he said:

> Essentially, we run nightly builds of OpenVSCode Server making it as upstream to VS Code as possible. In contrast to other forks, the whole scope of the project is to add a minimal set of changes (specifics about what we added can be found at https://github.com/gitpod-io/openvscode-server/)
> 
> The architecture we use powers both Gitpod and GitHub Codespaces. Several of the devs and organisations that have asked for our implementation & now have adopted OpenVSCode Server referenced the following issue at Code-Server https://github.com/cdr/code-server/issues/3835 

This is a good summary too: https://twitter.com/jolandgraf/status/1442850063953313805

**Reply by akosyakov** [reply]
Please also see a history behind OpenVSCode Server: https://www.gitpod.io/blog/cloud-ide-history it should explain on origins and scope of the project.
